custom_components/mqtt_discoverystream/__init__.py

```diff
@@ -361,12 +358,17 @@ def generate_discovery_topic(entity_id):
 
 
     async def my_async_subscribe_mqtt(hass, _):
-        await async_subscribe(hass, f"{base_topic}switch/+/set", message_received)
-        await async_subscribe(hass, f"{base_topic}light/+/set_light", message_received)
-        await async_subscribe(hass, f"{base_topic}input_boolean/+/set", message_received)
-        await async_subscribe(hass, f"{base_topic}button/+/set", message_received)
-        await async_subscribe(hass, f"{base_topic}script/+/set", message_received)
-        await async_subscribe(hass, f"{base_topic}input_button/+/set", message_received)
+        await mqtt.async_subscribe(hass, f"{base_topic}switch/+/set", message_received)
+        await mqtt.async_subscribe(hass, f"{base_topic}light/+/set_light", message_received)
+        await mqtt.async_subscribe(hass, f"{base_topic}input_boolean/+/set", message_received)
+        await mqtt.async_subscribe(hass, f"{base_topic}button/+/set", message_received)
+        await mqtt.async_subscribe(hass, f"{base_topic}script/+/set", message_received)
+        await mqtt.async_subscribe(hass, f"{base_topic}input_button/+/set", message_received)
+
+    # Make sure MQTT integration is enabled and the client is available
+    if not await mqtt.async_wait_for_mqtt_client(hass):
```

Comment:
Race condition handling. Make sure that MQTT is enabled and the client is connected before continuing on.

https://developers.home-assistant.io/docs/creating_integration_manifest/#mqtt